### PR TITLE
Swarm hash.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libdevcore/SHA3.h [hunk 1]
```diff
@@ -53,4 +53,12 @@ inline std::string keccak256(std::string const& _input, bool _isNibbles) { retur
 /// Calculate SHA3-256 MAC
 inline void keccak256mac(bytesConstRef _secret, bytesConstRef _plain, bytesRef _output) { keccak256(_secret.toBytes() + _plain.toBytes()).ref().populate(_output); }
 
+h256 swarmHash(bytes const& _data)
+{
+	bytes size(8);
+	for (size_t i = 0; i < 8; ++i)
```
Comment (axic) [inline libdevcore/SHA3.h]: > the files used will probably not exceed 4k, so I didn't bother implementing inner nodes

In that case please include an assert for size <= 4k.

**Comment by zelig** [comment]
LGTM mod assert <4k
Note that the hash will probably change with POC 0.4

**Comment by axic** [comment]
> Note that the hash will probably change with POC 0.4

Unfortunately this is the main problem with including Swarm/IPFS hashes in Solidity output.

**Comment by chriseth** [comment]
@axic I think it is still fine, as long as it is the hash of _something_.

**Comment by chriseth** [comment]
@axic This should now respect the strict specification.
@zeilig do you have any test vectors I could use?

**Comment by axic** [comment]
I think this would belong into its own file (or common), but not in SHA3.

**Comment by chriseth** [comment]
Updated.

**Review by axic — APPROVED** [review]
Looks good, but I'm still concerned the hashing algo will change with the next version of Swarm, rendering this output limited to old Swarm nodes.

**Comment by chriseth** [comment]
Sure, only so much we can do. Better to have some support now than no support at all.

**Pull request merged**